fieldpath/element.go

```diff
@@ -143,12 +143,12 @@ func (e PathElement) String() string {
 	case e.Key != nil:
 		strs := make([]string, len(*e.Key))
 		for i, k := range *e.Key {
-			strs[i] = fmt.Sprintf("%v=%v", k.Name, value.ToString(k.Value))
+			strs[i] = fmt.Sprintf("%v=%v", k.Name, value.AsString(k.Value))
```

Comment:
Here I think it's exactly NOT what we want. We don't want to cast the value into a string, but to print the value.

Reply:
Right, my bad.
value.ToString actually does something different than AsString.

This would be fitting the Stringer interface.
Shouldn't it be a String method for the Value interface or do you think having `AsString()` and `String()` would be error prone as it's to easy to use the wrong one?

Reply:
The problem is that `String` does not depend on the specific implementation of the string, so we can't make it part of the interface (or we could, but it'd be tedious).

Reply:
Ah, okay.
Then i'll just revert the two things you pointed out.

Reply:
Done